functions/src/storeTrackingData.ts

```diff
@@ -1,4 +1,100 @@
+export {};
+const { google } = require('googleapis');
+const functions = require('firebase-functions');
+const GenderApi = require('gender-api.com-client');
+
+function getAppDate(appDate: any) {
+  const monthNames = [
+    'Jan',
+    'Feb',
+    'Mar',
+    'Apr',
+    'May',
+    'Jun',
+    'Jul',
+    'Aug',
+    'Sep',
+    'Oct',
+    'Nov',
+    'Dec',
+  ];
+  const newdate =
+    appDate.getDate() +
+    '-' +
+    monthNames[appDate.getMonth()] +
+    '-' +
+    appDate.getFullYear();
+  return newdate;
+}
+
 exports.handler = async function(snap: any) {
   const email: string = snap.data()!.email;
-  console.log(email);
+  const name: string = snap.data()!.name;
+  const batch: string = snap.data()!.batch;
+  const userType: string = snap.data()!.userType;
+  const track: string = snap.data()!.track;
+  const source: string = snap.data()!.source;
+  const applicationTime: any = getAppDate(snap.createTime.toDate());
+
+  const OAuth2 = google.auth.OAuth2;
+  const clientID = functions.config().spreadsheet.clientid;
+  const clientSecret = functions.config().spreadsheet.clientsecret;
+  const refreshToken = functions.config().spreadsheet.refreshtoken;
+
+  const oauth2Client = new OAuth2(
+    clientID, //client Id
+    clientSecret, // Client Secret
+    'https://developers.google.com/oauthplayground' // Redirect URL
+  );
+
+  oauth2Client.setCredentials({
+    refresh_token: refreshToken,
+  });
+
+  function appendPromise(requestWithoutAuth: any) {
+    console.log('appendpromisee');
+    return new Promise((resolve, reject) => {
+      const sheets = google.sheets('v4');
+      const request = requestWithoutAuth;
+      return sheets.spreadsheets.values.append(
+        request,
+        (err: any, response: any) => {
+          if (err) {
+            console.log(`The API returned an error: ${err}`);
+            reject(err);
+            return;
+          }
+          resolve(response.data);
+          return;
+        }
+      );
+    });
+  }
+
+  // api used: https://gender-api.com/en/clients
+  const genderApiClient = new GenderApi.Client('VYdUEnwvvPsSKUeFwD');
```

Comment:
Seems like we shouldn't leak this one as well? It's probably connected to an account for a paid service and now anyone can use it

Reply:
Hey yup it's connected to some test account that I made without no personal/ sensitive information in it. Also it's not paid but has a quota. But yes you are right and it would have been better if it wasn't exposed like this. Thanks!